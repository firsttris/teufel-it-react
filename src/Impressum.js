import React, { Component } from 'react';
import Paper from './assets/images/backgrounds/paper.png';
import Logo from './logo.svg';
import './logo.css';

const blackBackgroundStyle = { backgroundImage: `url(${Paper})`, color: 'white' };

class Impressum extends Component {
  render() {
    const names = [
      "Inhaber:",
      "Tristan Teufel",
      "Werkstr.10a",
      "D-77815 Bühl"
    ];
    const sources = [
      "Quellenangaben für Bilder und Icons",
      "unsplash.com, shutterstock.com, fontawesome.io"
    ];
    return (
      <div className="row py-3" style={blackBackgroundStyle}>
        <div className="col-xs-12 col-md-4 col-lg-4">
          {names.map(row => <div key={row}>{row}</div>)}
        </div>
        <div className="col-xs-12 col-md-4 col-lg-4">
          <span>Build with <a href="https://facebook.github.io/react/" target="_blank" rel="noopener noreferrer" style={{ color: '#61dafb', textDecoration: 'none' }}><b>React</b></a></span>
          <img src={Logo} className="App-logo" alt="logo" />
          <div>
          <div>Quellen für Bilder und Icons:</div>
          <a href="http://unsplash.com" style={{textDecoration: 'none', color: '#61dafb' }} target="_blank" rel="noopener noreferrer" >unsplash.com </a>
          <a href="http://shutterstock.com" style={{textDecoration: 'none', color: '#61dafb' }} target="_blank" rel="noopener noreferrer" >shutterstock.com </a>
          <a href="http://fontawesome.io" style={{textDecoration: 'none', color: '#61dafb' }} target="_blank" rel="noopener noreferrer" >fontawesome.io</a>
          </div>
        </div>
        <div className="col-xs-12 col-md-4 col-lg-4">
          <div>Kontakt:</div>
          <div><a href="tel:+4917645744166" style={{textDecoration: 'none', color: '#61dafb' }}>+49 (0) 176 457 44 1 66</a></div>
          <div><a href="mailto:info@teufel-it.de" style={{textDecoration: 'none', color: '#61dafb' }}>info@teufel-it.de</a></div>
          <div>Finanzamt Baden-Baden 36216/14616</div>
        </div>
      </div>
    );
  }
}

export default Impressum;
