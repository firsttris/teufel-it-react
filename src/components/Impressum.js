import React, {Component} from 'react';
import {FormattedMessage} from 'react-intl';
import Paper from './../../public/images/backgrounds/paper.png';

const blackBackgroundStyle = {backgroundImage: `url(${Paper})`, color: 'white', textAlign: 'left', fontSize: 'small'};
const textStyle = {color: 'white'};
const fontSizeMedium = {fontSize: 'medium'};

class Impressum extends Component {
    render() {
        return (
            <div className="row" style={blackBackgroundStyle}>
                <div className="container py-3">
                    <p>© 2017 Teufel-IT</p>
                    <div className="row">
                        <div className="col-xs-12 col-md-3 col-lg-3 py-3">
                            <div style={fontSizeMedium}>
                                <FormattedMessage id="OWNER"/>
                            </div>
                            <br/>
                            <div>Tristan Teufel</div>
                            <div>Werkstr.10a</div>
                            <div>D-77815 Bühl</div>
                        </div>
                        <div className="col-xs-12 col-md-3 col-lg-3 py-3">
                            <div style={fontSizeMedium}>
                                <FormattedMessage id="SOURCE_CODE"/>
                            </div>
                            <br/>
                            <div>
                                <a style={textStyle}
                                   href="https://github.com/firsttris/teufel-it-react" rel="noopener noreferrer"
                                   target="_blank">
                                    <FormattedMessage id="GITHUB_PAGE_LINK"/><i className="fa fa-github"
                                                                                aria-hidden="true"/>
                                </a>
                            </div>
                        </div>
                        <div className="col-xs-12 col-md-3 col-lg-3 py-3">
                            <div style={fontSizeMedium}>
                                <FormattedMessage id="SOURCE_FOR_IMAGES"/>
                            </div>
                            <br/>
                            <div>
                                <a href="http://unsplash.com" style={textStyle} target="_blank"
                                   rel="noopener noreferrer">www.unsplash.com </a>
                            </div>
                            <div>
                                <a href="http://shutterstock.com" style={textStyle} target="_blank"
                                   rel="noopener noreferrer">www.shutterstock.com </a>
                            </div>
                            <div>
                                <a href="http://fontawesome.io" style={textStyle} target="_blank"
                                   rel="noopener noreferrer">www.fontawesome.io</a>
                            </div>
                        </div>
                        <div className="col-xs-12 col-md-3 col-lg-3 py-3">
                            <div style={fontSizeMedium}>
                                <FormattedMessage id="CONTACT_TITLE"/>
                            </div>
                            <br/>
                            <div><a href="tel:+4917645744166" style={textStyle}>+49 (0) 176 457 44 1 66</a></div>
                            <div><a href="mailto:info@teufel-it.de" style={textStyle}>info@teufel-it.de</a></div>
                            <div>Finanzamt Baden-Baden 36216/14624</div>
                        </div>
                    </div>
                </div>

            </div>
        );
    }
}

export default Impressum;
