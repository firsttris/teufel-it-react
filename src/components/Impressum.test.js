import React from 'react';
import Impressum from './Impressum';
import renderer from 'react-test-renderer';

test('Display Impressum', () => {
  const component = renderer.create(
    <Impressum></Impressum>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});