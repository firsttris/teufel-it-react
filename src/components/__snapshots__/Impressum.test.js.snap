// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Impressum 1`] = `
<div
  className="row py-3"
  style={
    Object {
      "backgroundImage": "url(paper.png)",
      "color": "white",
    }
  }
>
  <div
    className="col-xs-12 col-md-4 col-lg-4"
  >
    <div>
      Inhaber:
    </div>
    <div>
      Tristan Teufel
    </div>
    <div>
      Werkstr.10a
    </div>
    <div>
      D-77815 Bühl
    </div>
  </div>
  <div
    className="col-xs-12 col-md-4 col-lg-4"
  >
    <span>
      Build with 
      <a
        href="https://facebook.github.io/react/"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "#61DAFB",
            "textDecoration": "none",
          }
        }
        target="_blank"
      >
        <b>
          React
        </b>
      </a>
    </span>
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <div>
      <div>
        Quellen für Bilder und Icons:
      </div>
      <a
        href="http://unsplash.com"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "#61DAFB",
            "textDecoration": "none",
          }
        }
        target="_blank"
      >
        unsplash.com 
      </a>
      <a
        href="http://shutterstock.com"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "#61DAFB",
            "textDecoration": "none",
          }
        }
        target="_blank"
      >
        shutterstock.com 
      </a>
      <a
        href="http://fontawesome.io"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "#61DAFB",
            "textDecoration": "none",
          }
        }
        target="_blank"
      >
        fontawesome.io
      </a>
    </div>
  </div>
  <div
    className="col-xs-12 col-md-4 col-lg-4"
  >
    <div>
      Kontakt:
    </div>
    <div>
      <a
        href="tel:+4917645744166"
        style={
          Object {
            "color": "#61DAFB",
            "textDecoration": "none",
          }
        }
      >
        +49 (0) 176 457 44 1 66
      </a>
    </div>
    <div>
      <a
        href="mailto:info@teufel-it.de"
        style={
          Object {
            "color": "#61DAFB",
            "textDecoration": "none",
          }
        }
      >
        info@teufel-it.de
      </a>
    </div>
    <div>
      Finanzamt Baden-Baden 36216/14616
    </div>
  </div>
</div>
`;
